eZ/Publish/Core/Repository/ContentService.php

```diff
@@ -1917,15 +1918,25 @@ public function removeTranslation(ContentInfo $contentInfo, $languageCode)
             }
             // check if the specified translation is not the only one
             if (count($versionInfo->languageCodes) < 2) {
-                throw new BadStateException(
-                    '$languageCode',
-                    'Specified translation is the only one Content Object Version has'
-                );
+                $singleLangVersions[] = $versionInfo->versionNo;
+                continue;
             }
             // add version to the list
             $versions[] = $versionInfo->versionNo;
         }
 
+        if (!empty($singleLangVersions)) {
+            throw new BadStateException(
+                '$languageCode',
+                sprintf(
+                    'The Version(s): %s of the ContentId=%d have only one language %s. Remove these Version(s) before proceeding',
```

Comment:
Maybe also add, `\nHint: Command 'ezplatform:remove_content_translation' handles this for you, look to it for how this can be handled in custom code.`

Reply:
Now I have ugly-long line :) Refactored it a little bit in fixup 6c8df0f (not squashed). Hope this is fine.

Reply:
Yes, this is what I meant ;)